### PR TITLE
[WIP] Refactor MaskType Fix #482

### LINKED ISSUE #482: New mask types and change proposal
# Change proposal for faster new mask type implementation

- `MaskType` enum could provide an `UIBezierPath` using a function like this `func bezierPath(for rect: CGRect)`
- Then in `MaskDesignable` we ask the path and call the draw function only one time code

I think also a lot of code to create `UIBezierPath` must done in one `UIBezierPath` `extension` instead of `MaskDesignable` or `MaskType`
For instance
```swift
extension UIBezierPath {
    public convenience init(triangleInRect rect: CGRect) {
        // code from getTriangleBezierPath
    }
}
```
# New mask type

I have already a lot of `UIBezierPath` code to create the following path

- [x]  .heart ❤️ 
- [x]  .drop 💧 
- [x]  .gear(radius, cogs) ⚙️ 
- [x]  .ring(radius) 💍  
- [x]  .superEllipse 💠  https://en.wikipedia.org/wiki/Superellipse
- [x]  .plus ➕ 
- [x]  .crescent 🌙 

You can copy paste the list and checked the one you think that could be useful as mask layer

### PR DESCRIPTION
**Pull request by phimage**
To implement feature request #482

- [x] Move UIBezierPath creation code in one UIBezierPath extension

### CODE REVIEW

**Comment by phimage** [comment]
Even if the PR is WIP, you can review it
the change is explained in #482

**Comment by SD10** [comment]
Looks great! Very clean API 👍

File: IBAnimatable/UIBezierPathExtension.swift [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+extension UIBezierPath {
+
+  // MARK: - Circle
+
+  /**
+   Create a Bezier path for a circle shape.
+   
+   - Parameter bounds: The bounds of shape.
+   */
+  convenience init(circleIn bounds: CGRect) {
```
Comment (tbaranes) [inline IBAnimatable/UIBezierPathExtension.swift]: Should it be public?

**Review by tbaranes — COMMENTED** [review]
Nice idea! What about merge this first then make a PR per mask?

**Comment by phimage** [comment]
I think I will wait for #484 before doing a new PR with new branch
And I will have my new mask ready
I want to be sure that I can go on

and one PR by mask it's too many constraints, due to CHANGELOG for instance...
I ask in #482 to validate the wanted mask, if some are useless, please let me know

File: IBAnimatable/UIBezierPathExtension.swift (hunk 1 above)
Comment (phimage) [reply IBAnimatable/UIBezierPathExtension.swift]: I see that the XXXExtension.swift files contain public extension
but here is this no need to share externally this code

**Comment by tbaranes** [comment]
Since #484 is merged, you can rebased (or restart a new PR) and go ahead 👍

**Review by JakeLin — APPROVED** [review]
Cool, good start 👍

**Pull request closed without merging**